### PR TITLE
Minor: dynamic min/max bound in graph for test case details page

### PR DESCRIPTION
**Pull request by ShaileshParmar11**
### Describe your changes:
https://github.com/open-metadata/openmetadata-collate/issues/321

image

image

#
### Type of change:

- [ ] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+export const prepareChartData = ({
+  testCaseParameterValue,
+  testCaseResults,
+  entityThread,
+}: PrepareChartDataType) => {
+  const params =
+    testCaseParameterValue?.length >= 2 ? testCaseParameterValue : [];
+  const dataPoints: TestCaseChartDataType['data'] = [];
+  const yValues = params.reduce((acc, curr, i) => {
+    return { ...acc, [`y${i + 1}`]: parseInt(curr.value ?? '') };
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts:37]: `parseInt(curr.value ?? '')` here uses a string as a fallback for parseInt, resulting in NaN if the value is null or undefined.

File: openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  }, {});
+  testCaseResults.forEach((result) => {
+    const values = result.testResultValue?.reduce((acc, curr) => {
+      const value = round(parseFloat(curr.value ?? ''), 2) || 0;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts:41]: Same for parseFloat here.

File: openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts (hunk 1 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts:37]: Yes, it's expected to have such value so the chart doesn't draw anything!

File: openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts (hunk 2 above)
Comment (ShaileshParmar11) [reply openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/TestSummaryGraphUtils.ts:41]: here we have a fallback to 0

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**